File: src/schemas/market.py
from typing import List
from pydantic import BaseModel

class MarketIndex(BaseModel):
    symbol: str
    name: str
    price: float
    change: float
    change_percent: float

class TopMover(BaseModel):
    symbol: str
    price: float
    change: float
    change_percent: float

class MarketSummaryOut(BaseModel):
    indices: List[MarketIndex]
    top_gainers: List[TopMover]
    top_losers: List[TopMover]

File: web/types/market.ts
export interface MarketIndex {
    symbol: string;
    name: string;
    price: number;
    change: number;
    change_percent: number;
}

export interface TopMover {
    symbol: string;
    price: number;
    change: number;
    change_percent: number;
}

export interface MarketSummary {
    indices: MarketIndex[];
    top_gainers: TopMover[];
    top_losers: TopMover[];
}
